pkg/controller/tektoninstallation/tektoninstallation_controller.go

```diff
@@ -87,37 +105,111 @@ func (r *ReconcileTektonInstallation) Reconcile(request reconcile.Request) (reco
+		return reconcile.Result{}, r.statusUpdate(reqLogger, tektonInstallation, r.setStatusTektonInstallationFailed, "tekton installation failed with error: "+details)
+	default:
+		return reconcile.Result{}, r.statusUpdate(reqLogger, tektonInstallation, r.setStatusTektonUnknown, "tekton installation status is unknown")
 	}
-	return r.statusUpdate(logger, tektonInstallation, r.setStatusTektonSubscriptionReady, "")
 }
 
-func (r *ReconcileTektonInstallation) ensureTektonSubscription(logger logr.Logger, tektonInstallation *v1alpha1.TektonInstallation, ns string) error {
+func (r *ReconcileTektonInstallation) ensureTektonSubscription(logger logr.Logger, tektonInstallation *v1alpha1.TektonInstallation, ns string) (bool, error) {
 	sub := &olmv1alpha1.Subscription{}
 	err := r.client.Get(context.TODO(), types.NamespacedName{Namespace: ns, Name: SubscriptionName}, sub)
 	if err != nil && errors.IsNotFound(err) {
 		tektonSub := NewSubscription(ns)
 		logger.Info("Creating subscription for tekton", "Subscription.Namespace", ns, "Subscription.Name", tektonSub.Name)
 		if err := controllerutil.SetControllerReference(tektonInstallation, tektonSub, r.scheme); err != nil {
-			return err
+			return false, err
 		}
-		return r.client.Create(context.TODO(), tektonSub)
+		if err := r.client.Create(context.TODO(), tektonSub); err != nil {
+			return false, err
+		}
+		return true, nil
 	}
-	return err
+
+	return false, err
 }
 
-func (r *ReconcileTektonInstallation) setStatusTektonSubscriptionReady(tektonInstallation *v1alpha1.TektonInstallation, message string) error {
+func (r *ReconcileTektonInstallation) ensureWatchTektonConfig() (bool, error) {
+	r.mu.Lock()
+	defer r.mu.Unlock()
+	if r.watchTektonConfig != nil {
+		tektonCfg := &config.Config{}
+		if err := r.client.Get(context.TODO(), types.NamespacedName{Namespace: "default", Name: "default"}, tektonCfg); err != nil {
+			if meta.IsNoMatchError(err) {
+				log.Info("Tekton resource type does not exist yet", "message", err.Error())
+				return true, nil
+			}
+			if !errors.IsNotFound(err) { // ignore NotFound
+				log.Error(err, "Unexpected error while getting a TektonConfig to ensure a TektonConfig watcher can be created")
+				return false, err
+			}
+		}
+		if err := r.watchTektonConfig(); err != nil {
+			log.Error(err, "Unexpected error while creating a watcher on the Tekton resources", "message")
+			return false, err
+		}
+		log.Info("Added a watcher on the TektonConfig resources")
+		r.watchTektonConfig = nil // make sure watchTektonConfig() should NOT be called afterwards
+	}
+	log.Info("Watcher on the Tekton resources already added")
+	return false, nil
+}
+
+func getTektonConfigStatus(tektonCfg *config.Config) (config.InstallStatus, string) {
+	for _, conditions := range tektonCfg.Status.Conditions {
+		code := conditions.Code
+		if code == config.InstalledStatus || code == config.InstallingStatus || code == config.ErrorStatus {
+			return code, conditions.Details
+		}
+	}
+	return "unknown", ""
```

Comment:
couldn't we check if there is at least some condition and return details/code for that one?

Reply:
@MatousJobanek there are other codes, but i'm putting them all under the "unknown" category because they don't really mean much to us. according to the comments in config, the valid  values which indicate the installation of pipeline resources are the following:
error, installing , and installed https://github.com/tektoncd/operator/blob/master/pkg/apis/operator/v1alpha1/config_types.go#L26-L29

since the other statuses' don't provide any real value for us, i didn't see a point in returning a code or details...

see @alexeykazakov gist for instance: https://gist.github.com/alexeykazakov/4478e0de9dd4f2dfe8b5f2fe81418470